dcos_launch/terraform.py

```diff
@@ -0,0 +1,283 @@
+            binary = 'terraform'
+        return binary
+
+    def create(self):
+        try:
+            if os.path.exists(self.init_dir):
+                raise util.LauncherError('ClusterAlreadyExists', "Either the cluster you are trying to create is "
+                                                                 "already running or the init_dir you specified in your"
+                                                                 " config is already used by another active cluster.")
+            os.makedirs(self.init_dir)
+            # Check if Terraform is installed by running 'terraform version'. If that fails, install Terraform.
+            try:
+                subprocess.run([self.terraform_cmd(), 'version'], check=True, stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT)
+            except FileNotFoundError:
+                log.info('No Terraform installation detected. Terraform is now being installed.')
+                self._install_terraform()
+
+            if self.config['key_helper']:
+                self.key_helper()
+            else:
+                log.warning('WARNING: {}Since you did not set "key_helper: true" in your config, make sure your '
+                            'ssh-agent is running i.e. "eval `ssh-agent -s`" and that you have added your private key '
+                            'to it i.e. "ssh-add /path/to/key.pem". ssh-agent usage is specific to terraform, not '
+                            'dcos-launch.')
+
+            module = 'github.com/dcos/{}?ref={}/{}'.format(
+                'terraform-dcos-enterprise' if self.config['dcos-enterprise'] else 'terraform-dcos',
+                self.config['terraform_dcos_enterprise_version'] if self.config['dcos-enterprise'] else
+                self.config['terraform_dcos_version'], self.config['platform'])
+
+            # Converting our YAML config to the required format. You can find an example of that format in the
+            # Advance YAML Configuration" section here:
+            # https://github.com/mesosphere/terraform-dcos-enterprise/tree/master/aws
+            with open(self.cluster_profile_path, 'w') as file:
+                for k, v in self.config['terraform_config'].items():
+                    file.write(k + ' = ')
+                    if type(k) is dict:
+                        file.write('<<EOF\n{}\nEOF\n'.format(yaml.dump(v)))
+                    else:
+                        file.write('"{}"\n'.format(v))
+            subprocess.run([self.terraform_cmd(), 'init', '-from-module', module], cwd=self.init_dir,
+                           check=True, stderr=subprocess.STDOUT)
+            subprocess.run([self.terraform_cmd(), 'apply', '-auto-approve', '-var-file', self.cluster_profile_path],
+                           cwd=self.init_dir, check=True, stderr=subprocess.STDOUT, env=os.environ)
+        except Exception as e:
+            self.config['create_exception'] = e
```

Comment:
Should this go back in to `self.config`? Is it a config option or parameter? Why not use a separate field (`self.create_error`)?